core/src/main/scala/neotypes/query.scala

```diff
@@ -1,568 +1,479 @@
 
-  /** The query statement that will be executed. */
+sealed trait BaseQuery {
+  /** @param query The query statement that will be executed. */
   def query: String
 
-  /** The values substituted into the query statement. */
+  /** @param params The parameters that will be substituted into the query statement. */
   def params: Map[String, QueryParam]
 
   /** Creates a new query with an updated set of parameters.
     *
     * @note If params contains a key that is already present in the current query,
-    * the new one will override the previous one.
+    * the new value one will override the previous one.
     *
     * @param params QueryParams to be added.
-    * @return DeferredQuery with params added to existing params.
+    * @return a new query with params added to existing params.
     */
-  def withParams(params: Map[String, QueryParam]): BaseDeferredQuery[T]
-
-  protected def effectDriverTransaction[F[_], O](driver: Driver[F])
-                                                (tx: Transaction[F] => F[O]): F[O]
-  protected def effectDriverTransactionConfig[F[_], O](driver: Driver[F], config: TransactionConfig)
-                                                      (tx: Transaction[F] => F[O]): F[O]
-  protected def streamDriverTransaction[S[_], F[_]](driver: StreamingDriver[S, F])
-                                                   (st: StreamingTransaction[S, F] => S[T]): S[T]
-  protected def streamDriverTransactionConfig[S[_], F[_]](driver: StreamingDriver[S, F], config: TransactionConfig)
-                                                         (st: StreamingTransaction[S, F] => S[T]): S[T]
+  def withParams(params: Map[String, QueryParam]): BaseQuery
 
-  def collectAs[C](factory: Factory[T, C]): BaseCollectAsPartiallyApplied[T, C]
+  override final def toString(): String =
+    s"""${this.getClass.getSimpleName}:
+       |${query}
+       |params: ${params.view.map(param => s"${param._1}: ${param._2.underlying}").mkString(", ")}
+     """.stripMargin
+}
 
-  /** Executes the query and returns a List of values.
-    *
-    * @example
-    * {{{
-    * val result: F[List[(Person, Movie)]] =
-    *   "MATCH (p:Person {name: 'Charlize Theron'})-[]->(m:Movie) RETURN p,m"
-    *     .query[(Person, Movie)]
-    *     .list(driver)
-    * }}}
-    *
-    * @param driver neotypes driver.
-    * @param rm result mapper for type T.
-    * @tparam F effect type.
-    * @return An effectual value that will compute a List of T elements.
-    */
-  final def list[F[_], O](driver: Driver[F])
-                         (implicit rm: ResultMapper[T]): F[List[T]] =
-    effectDriverTransaction(driver)(tx => tx.list(query, params))
+/** Represents a query that doesn't produce results when executed. */
+final class ExecuteQuery private[neotypes] (
+    val query: String,
+    val params: Map[String, QueryParam]
+) extends BaseQuery {
+  override def withParams(params: Map[String, QueryParam]): ExecuteQuery =
```

Comment:
Well, to be totally honest, I am only leaving that because it already was there and I didn't want to break even more things.

But, in practice, I doubt anyone has used this.
Like, is not common to create a **cypher** query with a dummy set of parameters to then replace / change / manipulate them.